test_j1587_driver.py

```diff
@@ -0,0 +1,96 @@
+import multiprocessing
+import unittest
+
+import struct
+from hv_networks.J1708Driver import checksum
+from hv_networks.J1587Driver import J1587Driver
+from hv_networks.J1587Driver import J1708DriverFactory
+from hv_networks.J1587Driver import set_j1708_driver_factory
+
+
+# fake J1708Driver for testing
+class FakeJ1708Driver:
+    def __init__(self):
+        return
+
+    to_rx = list()
+
+    def add_to_rx(self, more_rx):
+        self.to_rx.extend(more_rx)
+
+    def read_message(self, checksum=False, timeout=0.5):
+        if len(self.to_rx) == 0:
+            return None
+        else:
+            message = self.to_rx.pop()
+            if checksum:  # NB: J1587Drive will always call with checksums=true
+                return message
+            else:
+                return message[:-1]
+
+    sent = multiprocessing.Queue()
+
+    def send_message(self, buf, has_check=False):
+        msg = buf
+        self.sent.put(msg)
+
+    def close(self):
+        return
+
+
+class FakeJ1708Factory(J1708DriverFactory):
+    def __init__(self):
+        super(FakeJ1708Factory, self).__init__()
+        self.memo_fake_driver = None
+
+    def make(self):
+        if self.memo_fake_driver is None:
+            self.memo_fake_driver = FakeJ1708Driver()
+        return self.memo_fake_driver
+
+    def clear(self):
+        self.memo_fake_driver = None
+
+
+class J1587TestClass(unittest.TestCase):
+    def setUp(self):
```

Comment:
bruh don't use camel case in method names

Reply:
lol I wish I could choose not too broseph. its the unittest.TestCase method override.